site-icon-maskable/site-icon-maskable-block-editor.php

```diff
@@ -0,0 +1,66 @@
+<?php
+/**
+ * Handles the 'site_icon_maskable' setting within a full-site-editing context
+ *
+ * @package PWA
+ */
+
+namespace PWA_WP;
+
+use Error;
+
+use function add_action;
+use function plugins_url;
+use function register_setting;
+use function wp_enqueue_script;
+use function wp_set_script_translations;
+
+/**
+ * [pwa__enqueue_block_editor_assets description]
+ *
+ * @package PWA
+ * @since   0.8.0-alpha
+ * @throws Error Fatals out when the block-filter files weren't built.
+ *
+ * @see  https://developer.wordpress.org/reference/hooks/enqueue_block_editor_assets/
+ */
+function enqueue_block_editor_assets__site_icon_maskable() {
```

Comment:
I think these functions could just as well be added to the `WP_Web_App_Manifest` class since that's where we're working with the site icon and `maskable`.

Nevertheless, this is somewhat similar to https://github.com/GoogleChromeLabs/pwa-wp/blob/develop/wp-includes/class-wp-customize-manager.php but the intention there was that the code would eventually be added to `WP_Customize_Manager`.

Reply:
>I think these functions could just as well be added to the WP_Web_App_Manifest class since that's where we're working with the site icon and maskable.

:+1:  I agree, as this is what I thought by myself, too. 

The `WP_Customize_Manager` would not be the best place, from my opinion, because the *Customizer* as a whole maybe not future-proof, in favor of the new *Site editor*.